src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -981,12 +986,12 @@ static void ValidateLength(string decoratorName, DecoratorSyntax decoratorSyntax
                 .WithAttachableType(TypeHelper.CreateTypeUnion(LanguageConstants.String, LanguageConstants.Object))
                 .WithEvaluator((_, targetType, targetObject) =>
                 {
-                    if (ReferenceEquals(targetType, LanguageConstants.String))
+                    if (targetType is PrimitiveType pt && pt.Name == "string")
```

Comment:
Worth creating a reusable method for this? Can we reference `LanguageConstants.TypeNameString` here?

Reply:
As an aside - we've gotten away with string comparisons for a long time, but I'm wondering if it's worth us just creating an enum field on `PrimitiveType` (`PrimitiveTypeKind`) to simplify these checks in the long run. I can create an issue for this if it's going to be a big refactor.

Reply:
Tracking in #8812